.github/workflows/protectSecurityRelevantCode.yml

```diff
@@ -0,0 +1,176 @@
+            local org=$1
+            local team=$2
+            gh api \
+              -H "Accept: application/vnd.github+json" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              "/orgs/$org/teams/$team/members" | jq -r '.[].login'
+          }
+
+          ORG_NAME='lifinance'
+          GROUP_NAME='InformationSecurityManager'
+
+          ##### get team members
+          INFORMATION_SECURITY_MEMBERS=$(getTeamMembers $ORG_NAME $GROUP_NAME)
+
+          echo "Team members of 'Information Security Manager' group: $INFORMATION_SECURITY_MEMBERS"
+
+          ##### store members in variable
+          echo -e "$INFORMATION_SECURITY_MEMBERS" > itSec_git_handles.txt
+
+      - name: Check approval of Information Security Manager
+        id: check-sec-mgr-approval
+        uses: actions/github-script@v7
+        if: env.CONTINUE == 'true'
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number || github.event.review.pull_request.number }}
+        with:
+          script: |
+            const fs = require('fs');
+            // ANSI escape codes for colors (used for colored output in Git action console)
+            const colors = {
+              reset: "\033[0m",
+              red: "\033[31m",
+              green: "\033[32m",
+              yellow: "\033[33m",
+            };
+
+            // Read git handles from file
+            const itSecHandlesFile = 'itSec_git_handles.txt';
+            const itSecHandles = fs.readFileSync(itSecHandlesFile, 'utf-8').split(/\r?\n/).filter(Boolean);
+            if (!itSecHandles) {
+              console.log(`${colors.red}Could not get the git handles of the InformationSecurityManager team.${colors.reset}`);
+              core.setFailed("Cannot read from InformationSecurityManager team.");
+              return;
+            }
+
+            const pullNumber = process.env.PR_NUMBER;
+            if (!pullNumber) {
+              console.log(`${colors.red}No PR number found in context.${colors.reset}`);
+              core.setFailed("PR number is missing.");
+              return;
+            }
+
+            // get all reviewers that have approved this PR
+            const { data: reviews } = await github.rest.pulls.listReviews({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: pullNumber,
+            });
+
+            // make sure that reviews are available
+            if(!reviews || reviews.length === 0) {
+              console.log(`${colors.red}Could not get reviewers of this PR from Github. Are there any reviews yet?${colors.reset}`);
+              console.log(`${colors.red}Check failed.${colors.reset}`);
+              core.setFailed("Required approval is missing");
+              return
+            }
+
+            // Filter to only include reviews that have "APPROVED" status
+            const approvedReviews = reviews.filter(review => review.state === 'APPROVED');
+
```

Comment:
@0xDEnYO did you check if that can be the case? If an approval was given but later it was rejected again or is not valid anymore because more changes were made in the PR?

Reply:
@maxklenk 

No I have not checked this, I cannot really check this alone. 
We dismiss stale approvals but I am not sure if they are still counted as valid by the app. My approach is to merge this and observe the behaviour and then adjust, if necessary